backend/app.py

```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
             body_token = body.get("token")
             # ensure the id and secret exist
             if body_token:
-                json_val = json.dumps(ec2_instances_mock_data)
+                json_val = json.dumps({"Nothing": "nothing"})
                 return Response(json_val, status=200, mimetype='application/json')
             return Response("Token missing", status=400, mimetype='application/text')
         else:
             return Response("Request body missing", status=400, mimetype='application/text')
 
+    @app.route('/api/google_authentication', methods=["POST"])
+    def google_authentication() -> Response:
+        """
+        A method to authenticate with google.
+
+        Returns
+        -------
+        Response
+            an http response
+        """
+
+        check_endpoint = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token="
+
+        body = request.get_json()
+
+        if body:
+            auth_email = body.get("email")
+            auth_token = body.get("token")
+            access_key = body.get("access_key")
+            secret_key = body.get("secret_key")
+
+            if all((auth_email, auth_token, access_key, secret_key)):
+                try:
+                    r = requests.get(check_endpoint + auth_token)
+                    data = r.json()
+
+                    if data.get("issued_to") and data.get("issued_to") == auth_email:
+                        token = str(uuid.uuid4())
+                        ttl = 128000
+
+                        db.insert('users', {
+                            'email': auth_email,
+                            'google_token': auth_token,
+                            'access_key': access_key,
+                            'secret_key': secret_key,
+                        })
+
+                        db.insert('access', {
+                            'user_id': auth_email,
+                            'token': token,
+                            'expires': datetime.now() + timedelta(seconds=ttl),
+                        })
+                        res = {'token': token, 'ttl': ttl}
+                        return Response(json.dumps(res), status=200, mimetype='application/json')
+                except Exception as e:
+                    res = {'error': e}
+                    return Response(json.dumps(res), status=400, mimetype='application/json')
+            else:
+                return Response(json.dumps({
+                    'message': f"Missing fields: email, token, access_key, or secret_key"
```

Comment:
Why did you make it an f-string if there are no variables?

Reply:
not sure will remove